webknossos-datastore/app/com/scalableminds/webknossos/datastore/dataformats/wkw/WKWBucketStreamSink.scala

```diff
@@ -1,32 +1,30 @@
 package com.scalableminds.webknossos.datastore.dataformats.wkw
 
 import java.io.DataOutputStream
-
 import com.scalableminds.util.geometry.Vec3Int
 import com.scalableminds.webknossos.datastore.models.BucketPosition
 import com.scalableminds.webknossos.datastore.models.datasource.DataLayer
 import com.scalableminds.util.io.{NamedFunctionStream, NamedStream}
+import com.scalableminds.webknossos.datastore.geometry.Vec3IntProto
 import com.scalableminds.webknossos.wrap.{BlockType, WKWFile, WKWHeader}
 
 import scala.collection.mutable
 import scala.concurrent.Future
 
 class WKWBucketStreamSink(val layer: DataLayer) extends WKWDataFormatHelper {
 
-  def apply(bucketStream: Iterator[(BucketPosition, Array[Byte])]): Iterator[NamedStream] = {
+  def apply(bucketStream: Iterator[(BucketPosition, Array[Byte])], mags: Seq[Vec3IntProto]): Iterator[NamedStream] = {
```

Comment:
I’d suggest passing a Vec3Int in here, (not the proto version). The proto version does not have all the nice convenience methods we use all around the scala code. You can have the caller extend the ProtoGeometryImplicits trait, which should make `vec3IntFromProto`  available (and since it is implicit, you might not even have to call it if the function signatures make the types unambiguous)